hashing/te-tag-query-python/all_in_one_lib/commands/base.py

```diff
@@ -0,0 +1,85 @@
+##!/usr/bin/env python
+
+"""
+Common helpers and libraries for the all-in-one command.
+
+Strongly consider moving classes and functions out of this file if it starts
+to fill up.
+"""
+
+import argparse
+import sys
+import typing as t
+
+from .. import common
+from ..dataset import Dataset
+
+
+class CommandError(Exception):
+    """Wrapper for exceptions which cause return codes"""
+
+    def __init__(self, message: str, returncode: int = 1) -> None:
+        super().__init__(message)
+        self.returncode = returncode
+
+
+class Command:
+    """
+    Simple wrapper around setting up commands for an argparse-based cli.
+    """
+
+    @classmethod
+    def add_command_to_subparser(cls, subparsers) -> None:
+        """
+        Shortcut for adding the command to the parser.
+
+        Propbably don't override.
+        """
+        command_ap = subparsers.add_parser(
+            cls.get_name(),
+            description=cls.get_description(),
+            help=cls.get_help(),
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+        )
+        command_ap.set_defaults(command_cls=cls)
+        cls.init_argparse(command_ap)
+
+    @classmethod
+    def init_argparse(cls, argparse: argparse.ArgumentParser) -> None:
+        """Program the command subparser for initFromNamespace"""
+        pass
```

Comment:
TODO: Check to see if **namespace.__dict__ works, if so, can save a piece of boilerplate.